wagtail/admin/tests/pages/test_edit_page.py

```diff
@@ -2014,7 +2014,7 @@ def test_page_edit_num_queries_as_superuser(self):
         # as when running it within the full test suite
         self.client.get(reverse("wagtailadmin_pages:edit", args=(self.event_page.id,)))
 
-        with self.assertNumQueries(35):
+        with self.assertNumQueries(36):
```

Comment:
Did you find out why this was necessary? Was there a query (maybe Locale-related) that is now redundant?

Reply:
Looks like an added query for locale here in both:

```diff
// test_page_edit_num_queries_as_editor
++ 33. SELECT "wagtailcore_locale"."id", "wagtailcore_locale"."language_code" FROM "wagtailcore_locale" ORDER BY "wagtailcore_locale"."language_code" ASC
// test_page_edit_num_queries_as_superuser
++ 29. SELECT "wagtailcore_locale"."id", "wagtailcore_locale"."language_code" FROM "wagtailcore_locale" ORDER BY "wagtailcore_locale"."language_code" ASC
```



<details>
<summary>After the change</summary>
```
======================================================================
test_page_edit_num_queries_as_editor (wagtail.admin.tests.pages.test_edit_page.TestPageEdit.test_page_edit_num_queries_as_editor)
----------------------------------------------------------------------

Captured queries were:
1. SELECT "django_session"."session_key", "django_session"."session_data", "django_session"."expire_date" FROM "django_session" WHERE ("django_session"."expire_date" > '2023-11-01 09:31:37.825320' AND "django_session"."session_key" = 'bb9y5i6pltqcw4rfxp7c50i1qj2gpxi9') LIMIT 21
2. SELECT "customuser_customuser"."password", "customuser_customuser"."last_login", "customuser_customuser"."is_superuser", "customuser_customuser"."identifier", "customuser_customuser"."username", "customuser_customuser"."email", "customuser_customuser"."is_staff", "customuser_customuser"."is_active", "customuser_customuser"."first_name", "customuser_customuser"."last_name", "customuser_customuser"."country", "customuser_customuser"."attachment" FROM "customuser_customuser" WHERE "customuser_customuser"."identifier" = -732135481 LIMIT 21
3. SELECT "django_content_type"."app_label", "auth_permission"."codename" FROM "auth_permission" INNER JOIN "customuser_customuser_user_permissions" ON ("auth_permission"."id" = "customuser_customuser_user_permissions"."permission_id") INNER JOIN "django_content_type" ON ("auth_permission"."content_type_id" = "django_content_type"."id") WHERE "customuser_customuser_user_permissions"."customuser_id" = -732135481
4. SELECT "django_content_type"."app_label", "auth_permission"."codename" FROM "auth_permission" INNER JOIN "auth_group_permissions" ON ("auth_permission"."id" = "auth_group_permissions"."permission_id") INNER JOIN "auth_group" ON ("auth_group_permissions"."group_id" = "auth_group"."id") INNER JOIN "customuser_customuser_groups" ON ("auth_group"."id" = "customuser_customuser_groups"."group_id") INNER JOIN "django_content_type" ON ("auth_permission"."content_type_id" = "django_content_type"."id") WHERE "customuser_customuser_groups"."customuser_id" = -732135481
5. SELECT "wagtailusers_userprofile"."id", "wagtailusers_userprofile"."user_id", "wagtailusers_userprofile"."submitted_notifications", "wagtailusers_userprofile"."approved_notifications", "wagtailusers_userprofile"."rejected_notifications", "wagtailusers_userprofile"."updated_comments_notifications", "wagtailusers_userprofile"."preferred_language", "wagtailusers_userprofile"."current_time_zone", "wagtailusers_userprofile"."avatar", "wagtailusers_userprofile"."dismissibles", "wagtailusers_userprofile"."theme" FROM "wagtailusers_userprofile" WHERE "wagtailusers_userprofile"."user_id" = -732135481 LIMIT 21
6. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."id" = 5 LIMIT 21
7. SELECT "wagtailcore_workflowstate"."id", "wagtailcore_workflowstate"."content_type_id", "wagtailcore_workflowstate"."base_content_type_id", "wagtailcore_workflowstate"."object_id", "wagtailcore_workflowstate"."workflow_id", "wagtailcore_workflowstate"."status", "wagtailcore_workflowstate"."created_at", "wagtailcore_workflowstate"."requested_by_id", "wagtailcore_workflowstate"."current_task_state_id", "wagtailcore_taskstate"."id", "wagtailcore_taskstate"."workflow_state_id", "wagtailcore_taskstate"."revision_id", "wagtailcore_taskstate"."task_id", "wagtailcore_taskstate"."status", "wagtailcore_taskstate"."started_at", "wagtailcore_taskstate"."finished_at", "wagtailcore_taskstate"."finished_by_id", "wagtailcore_taskstate"."comment", "wagtailcore_taskstate"."content_type_id", "wagtailcore_task"."id", "wagtailcore_task"."name", "wagtailcore_task"."content_type_id", "wagtailcore_task"."active" FROM "wagtailcore_workflowstate" LEFT OUTER JOIN "wagtailcore_taskstate" ON ("wagtailcore_workflowstate"."current_task_state_id" = "wagtailcore_taskstate"."id") LEFT OUTER JOIN "wagtailcore_task" ON ("wagtailcore_taskstate"."task_id" = "wagtailcore_task"."id") WHERE (("wagtailcore_workflowstate"."status" = 'in_progress' OR "wagtailcore_workflowstate"."status" = 'needs_changes') AND ("wagtailcore_workflowstate"."base_content_type_id" = 1 AND "wagtailcore_workflowstate"."object_id" IN ('5')))
8. SELECT "wagtailcore_revision"."id", "wagtailcore_revision"."content_type_id", "wagtailcore_revision"."base_content_type_id", "wagtailcore_revision"."object_id", "wagtailcore_revision"."submitted_for_moderation", "wagtailcore_revision"."created_at", "wagtailcore_revision"."user_id", "wagtailcore_revision"."object_str", "wagtailcore_revision"."content", "wagtailcore_revision"."approved_go_live_at" FROM "wagtailcore_revision" WHERE ("wagtailcore_revision"."content_type_id" = 38 AND "wagtailcore_revision"."object_id" = '5' AND "wagtailcore_revision"."approved_go_live_at" IS NOT NULL) ORDER BY "wagtailcore_revision"."id" ASC LIMIT 1
9. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id", "tests_eventpage"."page_ptr_id", "tests_eventpage"."date_from", "tests_eventpage"."date_to", "tests_eventpage"."time_from", "tests_eventpage"."time_to", "tests_eventpage"."audience", "tests_eventpage"."location", "tests_eventpage"."body", "tests_eventpage"."cost", "tests_eventpage"."signup_link", "tests_eventpage"."feed_image_id" FROM "tests_eventpage" INNER JOIN "wagtailcore_page" ON ("tests_eventpage"."page_ptr_id" = "wagtailcore_page"."id") WHERE "tests_eventpage"."page_ptr_id" = 5 LIMIT 21
10. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."path" = '00010001' LIMIT 21
11. SELECT "wagtailcore_grouppagepermission"."id", "wagtailcore_grouppagepermission"."group_id", "wagtailcore_grouppagepermission"."page_id", "wagtailcore_grouppagepermission"."permission_id", "wagtailcore_grouppagepermission"."permission_type", "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id", "auth_permission"."id", "auth_permission"."name", "auth_permission"."content_type_id", "auth_permission"."codename" FROM "wagtailcore_grouppagepermission" INNER JOIN "auth_group" ON ("wagtailcore_grouppagepermission"."group_id" = "auth_group"."id") INNER JOIN "customuser_customuser_groups" ON ("auth_group"."id" = "customuser_customuser_groups"."group_id") INNER JOIN "wagtailcore_page" ON ("wagtailcore_grouppagepermission"."page_id" = "wagtailcore_page"."id") LEFT OUTER JOIN "auth_permission" ON ("wagtailcore_grouppagepermission"."permission_id" = "auth_permission"."id") WHERE "customuser_customuser_groups"."customuser_id" = -732135481
12. SELECT "wagtailcore_pagesubscription"."id", "wagtailcore_pagesubscription"."user_id", "wagtailcore_pagesubscription"."page_id", "wagtailcore_pagesubscription"."comment_notifications" FROM "wagtailcore_pagesubscription" WHERE ("wagtailcore_pagesubscription"."page_id" = 5 AND "wagtailcore_pagesubscription"."user_id" = -732135481) LIMIT 21
13. SELECT "wagtailcore_workflowstate"."id", "wagtailcore_workflowstate"."content_type_id", "wagtailcore_workflowstate"."base_content_type_id", "wagtailcore_workflowstate"."object_id", "wagtailcore_workflowstate"."workflow_id", "wagtailcore_workflowstate"."status", "wagtailcore_workflowstate"."created_at", "wagtailcore_workflowstate"."requested_by_id", "wagtailcore_workflowstate"."current_task_state_id" FROM "wagtailcore_workflowstate" WHERE ("wagtailcore_workflowstate"."base_content_type_id" = 1 AND "wagtailcore_workflowstate"."object_id" = '5') ORDER BY "wagtailcore_workflowstate"."created_at" DESC LIMIT 1
14. SELECT "wagtailcore_comment"."id", "wagtailcore_comment"."page_id", "wagtailcore_comment"."user_id", "wagtailcore_comment"."text", "wagtailcore_comment"."contentpath", "wagtailcore_comment"."position", "wagtailcore_comment"."created_at", "wagtailcore_comment"."updated_at", "wagtailcore_comment"."revision_created_id", "wagtailcore_comment"."resolved_at", "wagtailcore_comment"."resolved_by_id" FROM "wagtailcore_comment" WHERE "wagtailcore_comment"."page_id" = 5
15. SELECT (1) AS "a" FROM "wagtailcore_page" INNER JOIN "wagtailcore_workflowpage" ON ("wagtailcore_page"."id" = "wagtailcore_workflowpage"."page_id") INNER JOIN "wagtailcore_workflow" ON ("wagtailcore_workflowpage"."workflow_id" = "wagtailcore_workflow"."id") WHERE ("wagtailcore_page"."path" IN ('0001', '00010001', '000100010003') AND "wagtailcore_workflowpage"."page_id" IS NOT NULL AND "wagtailcore_workflow"."active") LIMIT 1
16. SELECT (1) AS "a" FROM "wagtailcore_page" INNER JOIN "wagtailcore_workflowpage" ON ("wagtailcore_page"."id" = "wagtailcore_workflowpage"."page_id") INNER JOIN "wagtailcore_workflow" ON ("wagtailcore_workflowpage"."workflow_id" = "wagtailcore_workflow"."id") WHERE ("wagtailcore_page"."path" IN ('0001', '00010001', '000100010003') AND "wagtailcore_workflowpage"."page_id" IS NOT NULL AND "wagtailcore_workflow"."active") LIMIT 1
17. SELECT "tests_eventpagecarouselitem"."id", "tests_eventpagecarouselitem"."translation_key", "tests_eventpagecarouselitem"."locale_id", "tests_eventpagecarouselitem"."sort_order", "tests_eventpagecarouselitem"."link_external", "tests_eventpagecarouselitem"."link_page_id", "tests_eventpagecarouselitem"."link_document_id", "tests_eventpagecarouselitem"."image_id", "tests_eventpagecarouselitem"."embed_url", "tests_eventpagecarouselitem"."caption", "tests_eventpagecarouselitem"."page_id" FROM "tests_eventpagecarouselitem" WHERE "tests_eventpagecarouselitem"."page_id" = 5 ORDER BY "tests_eventpagecarouselitem"."sort_order" ASC
18. SELECT "tests_eventpagespeaker"."id", "tests_eventpagespeaker"."translation_key", "tests_eventpagespeaker"."locale_id", "tests_eventpagespeaker"."sort_order", "tests_eventpagespeaker"."link_external", "tests_eventpagespeaker"."link_page_id", "tests_eventpagespeaker"."link_document_id", "tests_eventpagespeaker"."page_id", "tests_eventpagespeaker"."first_name", "tests_eventpagespeaker"."last_name", "tests_eventpagespeaker"."image_id" FROM "tests_eventpagespeaker" WHERE "tests_eventpagespeaker"."page_id" = 5 ORDER BY "tests_eventpagespeaker"."sort_order" ASC
19. SELECT "tests_eventpagerelatedlink"."id", "tests_eventpagerelatedlink"."translation_key", "tests_eventpagerelatedlink"."locale_id", "tests_eventpagerelatedlink"."sort_order", "tests_eventpagerelatedlink"."link_external", "tests_eventpagerelatedlink"."link_page_id", "tests_eventpagerelatedlink"."link_document_id", "tests_eventpagerelatedlink"."title", "tests_eventpagerelatedlink"."page_id" FROM "tests_eventpagerelatedlink" WHERE "tests_eventpagerelatedlink"."page_id" = 5 ORDER BY "tests_eventpagerelatedlink"."sort_order" ASC
20. SELECT "tests_headcountrelatedmodelusingpk"."custom_id", "tests_headcountrelatedmodelusingpk"."event_page_id", "tests_headcountrelatedmodelusingpk"."head_count" FROM "tests_headcountrelatedmodelusingpk" WHERE "tests_headcountrelatedmodelusingpk"."event_page_id" = 5 ORDER BY "tests_headcountrelatedmodelusingpk"."custom_id" ASC
21. SELECT "wagtailcore_locale"."id", "wagtailcore_locale"."language_code" FROM "wagtailcore_locale" ORDER BY "wagtailcore_locale"."language_code" ASC
22. SELECT "wagtailcore_groupcollectionpermission"."id", "wagtailcore_groupcollectionpermission"."group_id", "wagtailcore_groupcollectionpermission"."collection_id", "wagtailcore_groupcollectionpermission"."permission_id", "wagtailcore_collection"."id", "wagtailcore_collection"."path", "wagtailcore_collection"."depth", "wagtailcore_collection"."numchild", "wagtailcore_collection"."name", "auth_permission"."id", "auth_permission"."name", "auth_permission"."content_type_id", "auth_permission"."codename" FROM "wagtailcore_groupcollectionpermission" INNER JOIN "auth_group" ON ("wagtailcore_groupcollectionpermission"."group_id" = "auth_group"."id") INNER JOIN "customuser_customuser_groups" ON ("auth_group"."id" = "customuser_customuser_groups"."group_id") INNER JOIN "wagtailcore_collection" ON ("wagtailcore_groupcollectionpermission"."collection_id" = "wagtailcore_collection"."id") INNER JOIN "auth_permission" ON ("wagtailcore_groupcollectionpermission"."permission_id" = "auth_permission"."id") WHERE "customuser_customuser_groups"."customuser_id" = -732135481
23. SELECT (1) AS "a" FROM "wagtailcore_page" WHERE ((("wagtailcore_page"."path" LIKE '0001%' ESCAPE '\' AND "wagtailcore_page"."depth" >= 1 AND "wagtailcore_page"."owner_id" = -732135481) OR ("wagtailcore_page"."path" LIKE '0001%' ESCAPE '\' AND "wagtailcore_page"."depth" >= 1)) AND "wagtailcore_page"."content_type_id" IN (189, 40, 79, 67, 78, 75, 65)) LIMIT 1
24. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."path" = '0001' LIMIT 21
25. SELECT "wagtailcore_page"."id", "wagtailcore_page"."content_type_id" FROM "wagtailcore_page" WHERE ("wagtailcore_page"."path" IN ('0001', '00010001', '000100010003') AND "wagtailcore_page"."path" LIKE '0001%' ESCAPE '\' AND "wagtailcore_page"."depth" >= 1) ORDER BY "wagtailcore_page"."path" ASC
26. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."id" IN (1, 2) ORDER BY "wagtailcore_page"."path" ASC
27. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id", "tests_eventpage"."page_ptr_id", "tests_eventpage"."date_from", "tests_eventpage"."date_to", "tests_eventpage"."time_from", "tests_eventpage"."time_to", "tests_eventpage"."audience", "tests_eventpage"."location", "tests_eventpage"."body", "tests_eventpage"."cost", "tests_eventpage"."signup_link", "tests_eventpage"."feed_image_id" FROM "tests_eventpage" INNER JOIN "wagtailcore_page" ON ("tests_eventpage"."page_ptr_id" = "wagtailcore_page"."id") WHERE "tests_eventpage"."page_ptr_id" IN (5) ORDER BY "wagtailcore_page"."path" ASC
28. SELECT (1) AS "a" FROM "wagtailcore_page" WHERE ("wagtailcore_page"."path" LIKE '000100010003%' ESCAPE '\' AND "wagtailcore_page"."depth" >= 3 AND "wagtailcore_page"."live") LIMIT 1
29. SELECT (1) AS "a" FROM "wagtailcore_page" WHERE ("wagtailcore_page"."path" LIKE '000100010003%' ESCAPE '\' AND "wagtailcore_page"."depth" >= 3 AND "wagtailcore_page"."live") LIMIT 1
30. SELECT "cache_key", "value", "expires" FROM "cache" WHERE "cache_key" IN (':2:wagtail_site_root_paths')
31. SELECT "wagtailcore_page"."path" FROM "wagtailcore_pageviewrestriction" INNER JOIN "wagtailcore_page" ON ("wagtailcore_pageviewrestriction"."page_id" = "wagtailcore_page"."id")
32. SELECT COUNT(*) FROM (SELECT DISTINCT "wagtailcore_referenceindex"."base_content_type_id" AS "col1", "wagtailcore_referenceindex"."object_id" AS "col2" FROM "wagtailcore_referenceindex" WHERE ("wagtailcore_referenceindex"."to_content_type_id" = 1 AND "wagtailcore_referenceindex"."to_object_id" = '5')) subquery
33. SELECT "wagtailcore_locale"."id", "wagtailcore_locale"."language_code" FROM "wagtailcore_locale" ORDER BY "wagtailcore_locale"."language_code" ASC
35. SELECT "tests_eventcategory"."id", "tests_eventcategory"."translation_key", "tests_eventcategory"."locale_id", "tests_eventcategory"."name" FROM "tests_eventcategory" INNER JOIN "tests_eventpage_categories" ON ("tests_eventcategory"."id" = "tests_eventpage_categories"."eventcategory_id") WHERE "tests_eventpage_categories"."eventpage_id" = 5
36. SELECT "tests_eventcategory"."id", "tests_eventcategory"."translation_key", "tests_eventcategory"."locale_id", "tests_eventcategory"."name" FROM "tests_eventcategory"
37. SELECT "customuser_customuser"."password", "customuser_customuser"."last_login", "customuser_customuser"."is_superuser", "customuser_customuser"."identifier", "customuser_customuser"."username", "customuser_customuser"."email", "customuser_customuser"."is_staff", "customuser_customuser"."is_active", "customuser_customuser"."first_name", "customuser_customuser"."last_name", "customuser_customuser"."country", "customuser_customuser"."attachment", "wagtailusers_userprofile"."id", "wagtailusers_userprofile"."user_id", "wagtailusers_userprofile"."submitted_notifications", "wagtailusers_userprofile"."approved_notifications", "wagtailusers_userprofile"."rejected_notifications", "wagtailusers_userprofile"."updated_comments_notifications", "wagtailusers_userprofile"."preferred_language", "wagtailusers_userprofile"."current_time_zone", "wagtailusers_userprofile"."avatar", "wagtailusers_userprofile"."dismissibles", "wagtailusers_userprofile"."theme" FROM "customuser_customuser" LEFT OUTER JOIN "wagtailusers_userprofile" ON ("customuser_customuser"."identifier" = "wagtailusers_userprofile"."user_id") WHERE "customuser_customuser"."identifier" IN (-732135481)
38. SELECT "wagtailcore_workflowpage"."page_id", "wagtailcore_workflowpage"."workflow_id" FROM "wagtailcore_workflowpage" WHERE "wagtailcore_workflowpage"."page_id" = 5 LIMIT 21
39. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" INNER JOIN "wagtailcore_workflowpage" ON ("wagtailcore_page"."id" = "wagtailcore_workflowpage"."page_id") INNER JOIN "wagtailcore_workflow" ON ("wagtailcore_workflowpage"."workflow_id" = "wagtailcore_workflow"."id") WHERE ("wagtailcore_page"."path" IN ('0001', '00010001', '000100010003') AND NOT ("wagtailcore_page"."id" = 5) AND "wagtailcore_workflowpage"."page_id" IS NOT NULL AND "wagtailcore_workflow"."active") ORDER BY "wagtailcore_page"."depth" DESC LIMIT 1
40. SELECT "wagtailcore_workflowpage"."page_id", "wagtailcore_workflowpage"."workflow_id" FROM "wagtailcore_workflowpage" WHERE "wagtailcore_workflowpage"."page_id" = 1 LIMIT 21
41. SELECT "wagtailcore_workflow"."id", "wagtailcore_workflow"."name", "wagtailcore_workflow"."active" FROM "wagtailcore_workflow" WHERE "wagtailcore_workflow"."id" = 1 LIMIT 21

======================================================================
test_page_edit_num_queries_as_superuser (wagtail.admin.tests.pages.test_edit_page.TestPageEdit.test_page_edit_num_queries_as_superuser)
----------------------------------------------------------------------

Captured queries were:
1. SELECT "django_session"."session_key", "django_session"."session_data", "django_session"."expire_date" FROM "django_session" WHERE ("django_session"."expire_date" > '2023-11-01 09:31:47.108713' AND "django_session"."session_key" = 'vdrko9c5kbmvcul7k023v1hy9nbtjiyj') LIMIT 21
2. SELECT "customuser_customuser"."password", "customuser_customuser"."last_login", "customuser_customuser"."is_superuser", "customuser_customuser"."identifier", "customuser_customuser"."username", "customuser_customuser"."email", "customuser_customuser"."is_staff", "customuser_customuser"."is_active", "customuser_customuser"."first_name", "customuser_customuser"."last_name", "customuser_customuser"."country", "customuser_customuser"."attachment" FROM "customuser_customuser" WHERE "customuser_customuser"."identifier" = 1785040285 LIMIT 21
3. SELECT "wagtailusers_userprofile"."id", "wagtailusers_userprofile"."user_id", "wagtailusers_userprofile"."submitted_notifications", "wagtailusers_userprofile"."approved_notifications", "wagtailusers_userprofile"."rejected_notifications", "wagtailusers_userprofile"."updated_comments_notifications", "wagtailusers_userprofile"."preferred_language", "wagtailusers_userprofile"."current_time_zone", "wagtailusers_userprofile"."avatar", "wagtailusers_userprofile"."dismissibles", "wagtailusers_userprofile"."theme" FROM "wagtailusers_userprofile" WHERE "wagtailusers_userprofile"."user_id" = 1785040285 LIMIT 21
4. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."id" = 5 LIMIT 21
5. SELECT "wagtailcore_workflowstate"."id", "wagtailcore_workflowstate"."content_type_id", "wagtailcore_workflowstate"."base_content_type_id", "wagtailcore_workflowstate"."object_id", "wagtailcore_workflowstate"."workflow_id", "wagtailcore_workflowstate"."status", "wagtailcore_workflowstate"."created_at", "wagtailcore_workflowstate"."requested_by_id", "wagtailcore_workflowstate"."current_task_state_id", "wagtailcore_taskstate"."id", "wagtailcore_taskstate"."workflow_state_id", "wagtailcore_taskstate"."revision_id", "wagtailcore_taskstate"."task_id", "wagtailcore_taskstate"."status", "wagtailcore_taskstate"."started_at", "wagtailcore_taskstate"."finished_at", "wagtailcore_taskstate"."finished_by_id", "wagtailcore_taskstate"."comment", "wagtailcore_taskstate"."content_type_id", "wagtailcore_task"."id", "wagtailcore_task"."name", "wagtailcore_task"."content_type_id", "wagtailcore_task"."active" FROM "wagtailcore_workflowstate" LEFT OUTER JOIN "wagtailcore_taskstate" ON ("wagtailcore_workflowstate"."current_task_state_id" = "wagtailcore_taskstate"."id") LEFT OUTER JOIN "wagtailcore_task" ON ("wagtailcore_taskstate"."task_id" = "wagtailcore_task"."id") WHERE (("wagtailcore_workflowstate"."status" = 'in_progress' OR "wagtailcore_workflowstate"."status" = 'needs_changes') AND ("wagtailcore_workflowstate"."base_content_type_id" = 1 AND "wagtailcore_workflowstate"."object_id" IN ('5')))
6. SELECT "wagtailcore_revision"."id", "wagtailcore_revision"."content_type_id", "wagtailcore_revision"."base_content_type_id", "wagtailcore_revision"."object_id", "wagtailcore_revision"."submitted_for_moderation", "wagtailcore_revision"."created_at", "wagtailcore_revision"."user_id", "wagtailcore_revision"."object_str", "wagtailcore_revision"."content", "wagtailcore_revision"."approved_go_live_at" FROM "wagtailcore_revision" WHERE ("wagtailcore_revision"."content_type_id" = 38 AND "wagtailcore_revision"."object_id" = '5' AND "wagtailcore_revision"."approved_go_live_at" IS NOT NULL) ORDER BY "wagtailcore_revision"."id" ASC LIMIT 1
7. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id", "tests_eventpage"."page_ptr_id", "tests_eventpage"."date_from", "tests_eventpage"."date_to", "tests_eventpage"."time_from", "tests_eventpage"."time_to", "tests_eventpage"."audience", "tests_eventpage"."location", "tests_eventpage"."body", "tests_eventpage"."cost", "tests_eventpage"."signup_link", "tests_eventpage"."feed_image_id" FROM "tests_eventpage" INNER JOIN "wagtailcore_page" ON ("tests_eventpage"."page_ptr_id" = "wagtailcore_page"."id") WHERE "tests_eventpage"."page_ptr_id" = 5 LIMIT 21
8. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."path" = '00010001' LIMIT 21
9. SELECT "wagtailcore_pagesubscription"."id", "wagtailcore_pagesubscription"."user_id", "wagtailcore_pagesubscription"."page_id", "wagtailcore_pagesubscription"."comment_notifications" FROM "wagtailcore_pagesubscription" WHERE ("wagtailcore_pagesubscription"."page_id" = 5 AND "wagtailcore_pagesubscription"."user_id" = 1785040285) LIMIT 21
10. SELECT "wagtailcore_workflowstate"."id", "wagtailcore_workflowstate"."content_type_id", "wagtailcore_workflowstate"."base_content_type_id", "wagtailcore_workflowstate"."object_id", "wagtailcore_workflowstate"."workflow_id", "wagtailcore_workflowstate"."status", "wagtailcore_workflowstate"."created_at", "wagtailcore_workflowstate"."requested_by_id", "wagtailcore_workflowstate"."current_task_state_id" FROM "wagtailcore_workflowstate" WHERE ("wagtailcore_workflowstate"."base_content_type_id" = 1 AND "wagtailcore_workflowstate"."object_id" = '5') ORDER BY "wagtailcore_workflowstate"."created_at" DESC LIMIT 1
11. SELECT "wagtailcore_comment"."id", "wagtailcore_comment"."page_id", "wagtailcore_comment"."user_id", "wagtailcore_comment"."text", "wagtailcore_comment"."contentpath", "wagtailcore_comment"."position", "wagtailcore_comment"."created_at", "wagtailcore_comment"."updated_at", "wagtailcore_comment"."revision_created_id", "wagtailcore_comment"."resolved_at", "wagtailcore_comment"."resolved_by_id" FROM "wagtailcore_comment" WHERE "wagtailcore_comment"."page_id" = 5
12. SELECT (1) AS "a" FROM "wagtailcore_page" INNER JOIN "wagtailcore_workflowpage" ON ("wagtailcore_page"."id" = "wagtailcore_workflowpage"."page_id") INNER JOIN "wagtailcore_workflow" ON ("wagtailcore_workflowpage"."workflow_id" = "wagtailcore_workflow"."id") WHERE ("wagtailcore_page"."path" IN ('0001', '00010001', '000100010003') AND "wagtailcore_workflowpage"."page_id" IS NOT NULL AND "wagtailcore_workflow"."active") LIMIT 1
13. SELECT (1) AS "a" FROM "wagtailcore_page" INNER JOIN "wagtailcore_workflowpage" ON ("wagtailcore_page"."id" = "wagtailcore_workflowpage"."page_id") INNER JOIN "wagtailcore_workflow" ON ("wagtailcore_workflowpage"."workflow_id" = "wagtailcore_workflow"."id") WHERE ("wagtailcore_page"."path" IN ('0001', '00010001', '000100010003') AND "wagtailcore_workflowpage"."page_id" IS NOT NULL AND "wagtailcore_workflow"."active") LIMIT 1
14. SELECT "tests_eventpagecarouselitem"."id", "tests_eventpagecarouselitem"."translation_key", "tests_eventpagecarouselitem"."locale_id", "tests_eventpagecarouselitem"."sort_order", "tests_eventpagecarouselitem"."link_external", "tests_eventpagecarouselitem"."link_page_id", "tests_eventpagecarouselitem"."link_document_id", "tests_eventpagecarouselitem"."image_id", "tests_eventpagecarouselitem"."embed_url", "tests_eventpagecarouselitem"."caption", "tests_eventpagecarouselitem"."page_id" FROM "tests_eventpagecarouselitem" WHERE "tests_eventpagecarouselitem"."page_id" = 5 ORDER BY "tests_eventpagecarouselitem"."sort_order" ASC
15. SELECT "tests_eventpagespeaker"."id", "tests_eventpagespeaker"."translation_key", "tests_eventpagespeaker"."locale_id", "tests_eventpagespeaker"."sort_order", "tests_eventpagespeaker"."link_external", "tests_eventpagespeaker"."link_page_id", "tests_eventpagespeaker"."link_document_id", "tests_eventpagespeaker"."page_id", "tests_eventpagespeaker"."first_name", "tests_eventpagespeaker"."last_name", "tests_eventpagespeaker"."image_id" FROM "tests_eventpagespeaker" WHERE "tests_eventpagespeaker"."page_id" = 5 ORDER BY "tests_eventpagespeaker"."sort_order" ASC
16. SELECT "tests_eventpagerelatedlink"."id", "tests_eventpagerelatedlink"."translation_key", "tests_eventpagerelatedlink"."locale_id", "tests_eventpagerelatedlink"."sort_order", "tests_eventpagerelatedlink"."link_external", "tests_eventpagerelatedlink"."link_page_id", "tests_eventpagerelatedlink"."link_document_id", "tests_eventpagerelatedlink"."title", "tests_eventpagerelatedlink"."page_id" FROM "tests_eventpagerelatedlink" WHERE "tests_eventpagerelatedlink"."page_id" = 5 ORDER BY "tests_eventpagerelatedlink"."sort_order" ASC
17. SELECT "tests_headcountrelatedmodelusingpk"."custom_id", "tests_headcountrelatedmodelusingpk"."event_page_id", "tests_headcountrelatedmodelusingpk"."head_count" FROM "tests_headcountrelatedmodelusingpk" WHERE "tests_headcountrelatedmodelusingpk"."event_page_id" = 5 ORDER BY "tests_headcountrelatedmodelusingpk"."custom_id" ASC
18. SELECT "wagtailcore_locale"."id", "wagtailcore_locale"."language_code" FROM "wagtailcore_locale" ORDER BY "wagtailcore_locale"."language_code" ASC
19. SELECT (1) AS "a" FROM "wagtailcore_page" WHERE "wagtailcore_page"."content_type_id" IN (189, 40, 79, 67, 78, 75, 65) LIMIT 1
20. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."depth" = 1 ORDER BY "wagtailcore_page"."path" ASC
21. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."path" = '0001' LIMIT 21
22. SELECT "wagtailcore_page"."id", "wagtailcore_page"."content_type_id" FROM "wagtailcore_page" WHERE ("wagtailcore_page"."path" IN ('0001', '00010001', '000100010003') AND "wagtailcore_page"."path" LIKE '0001%' ESCAPE '\' AND "wagtailcore_page"."depth" >= 1) ORDER BY "wagtailcore_page"."path" ASC
23. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."id" IN (1, 2) ORDER BY "wagtailcore_page"."path" ASC
24. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id", "tests_eventpage"."page_ptr_id", "tests_eventpage"."date_from", "tests_eventpage"."date_to", "tests_eventpage"."time_from", "tests_eventpage"."time_to", "tests_eventpage"."audience", "tests_eventpage"."location", "tests_eventpage"."body", "tests_eventpage"."cost", "tests_eventpage"."signup_link", "tests_eventpage"."feed_image_id" FROM "tests_eventpage" INNER JOIN "wagtailcore_page" ON ("tests_eventpage"."page_ptr_id" = "wagtailcore_page"."id") WHERE "tests_eventpage"."page_ptr_id" IN (5) ORDER BY "wagtailcore_page"."path" ASC
25. SELECT (1) AS "a" FROM "wagtailcore_locale" WHERE NOT ("wagtailcore_locale"."id" IN (SELECT U1."locale_id" FROM "tests_eventpage" U0 INNER JOIN "wagtailcore_page" U1 ON (U0."page_ptr_id" = U1."id") WHERE U1."translation_key" = '3edd5a1148004ca3bbc3b2f0334f184e')) LIMIT 1
26. SELECT "cache_key", "value", "expires" FROM "cache" WHERE "cache_key" IN (':2:wagtail_site_root_paths')
27. SELECT "wagtailcore_page"."path" FROM "wagtailcore_pageviewrestriction" INNER JOIN "wagtailcore_page" ON ("wagtailcore_pageviewrestriction"."page_id" = "wagtailcore_page"."id")
28. SELECT COUNT(*) FROM (SELECT DISTINCT "wagtailcore_referenceindex"."base_content_type_id" AS "col1", "wagtailcore_referenceindex"."object_id" AS "col2" FROM "wagtailcore_referenceindex" WHERE ("wagtailcore_referenceindex"."to_content_type_id" = 1 AND "wagtailcore_referenceindex"."to_object_id" = '5')) subquery
29. SELECT "wagtailcore_locale"."id", "wagtailcore_locale"."language_code" FROM "wagtailcore_locale" ORDER BY "wagtailcore_locale"."language_code" ASC
30. SELECT "tests_eventcategory"."id", "tests_eventcategory"."translation_key", "tests_eventcategory"."locale_id", "tests_eventcategory"."name" FROM "tests_eventcategory" INNER JOIN "tests_eventpage_categories" ON ("tests_eventcategory"."id" = "tests_eventpage_categories"."eventcategory_id") WHERE "tests_eventpage_categories"."eventpage_id" = 5
31. SELECT "tests_eventcategory"."id", "tests_eventcategory"."translation_key", "tests_eventcategory"."locale_id", "tests_eventcategory"."name" FROM "tests_eventcategory"
32. SELECT "customuser_customuser"."password", "customuser_customuser"."last_login", "customuser_customuser"."is_superuser", "customuser_customuser"."identifier", "customuser_customuser"."username", "customuser_customuser"."email", "customuser_customuser"."is_staff", "customuser_customuser"."is_active", "customuser_customuser"."first_name", "customuser_customuser"."last_name", "customuser_customuser"."country", "customuser_customuser"."attachment", "wagtailusers_userprofile"."id", "wagtailusers_userprofile"."user_id", "wagtailusers_userprofile"."submitted_notifications", "wagtailusers_userprofile"."approved_notifications", "wagtailusers_userprofile"."rejected_notifications", "wagtailusers_userprofile"."updated_comments_notifications", "wagtailusers_userprofile"."preferred_language", "wagtailusers_userprofile"."current_time_zone", "wagtailusers_userprofile"."avatar", "wagtailusers_userprofile"."dismissibles", "wagtailusers_userprofile"."theme" FROM "customuser_customuser" LEFT OUTER JOIN "wagtailusers_userprofile" ON ("customuser_customuser"."identifier" = "wagtailusers_userprofile"."user_id") WHERE "customuser_customuser"."identifier" IN (1785040285)
33. SELECT "wagtailcore_workflowpage"."page_id", "wagtailcore_workflowpage"."workflow_id" FROM "wagtailcore_workflowpage" WHERE "wagtailcore_workflowpage"."page_id" = 5 LIMIT 21
35. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" INNER JOIN "wagtailcore_workflowpage" ON ("wagtailcore_page"."id" = "wagtailcore_workflowpage"."page_id") INNER JOIN "wagtailcore_workflow" ON ("wagtailcore_workflowpage"."workflow_id" = "wagtailcore_workflow"."id") WHERE ("wagtailcore_page"."path" IN ('0001', '00010001', '000100010003') AND NOT ("wagtailcore_page"."id" = 5) AND "wagtailcore_workflowpage"."page_id" IS NOT NULL AND "wagtailcore_workflow"."active") ORDER BY "wagtailcore_page"."depth" DESC LIMIT 1
36. SELECT "wagtailcore_workflowpage"."page_id", "wagtailcore_workflowpage"."workflow_id" FROM "wagtailcore_workflowpage" WHERE "wagtailcore_workflowpage"."page_id" = 1 LIMIT 21
37. SELECT "wagtailcore_workflow"."id", "wagtailcore_workflow"."name", "wagtailcore_workflow"."active" FROM "wagtailcore_workflow" WHERE "wagtailcore_workflow"."id" = 1 LIMIT 21
```
</details>

<details>
<summary>Before the change</summary>

```
======================================================================
test_page_edit_num_queries_as_editor (wagtail.admin.tests.pages.test_edit_page.TestPageEdit.test_page_edit_num_queries_as_editor)
----------------------------------------------------------------------

Captured queries were:
1. SELECT "django_session"."session_key", "django_session"."session_data", "django_session"."expire_date" FROM "django_session" WHERE ("django_session"."expire_date" > '2023-11-01 09:23:42.441108' AND "django_session"."session_key" = 'xnha92ov0eunx0c3nx4t1nwzf0lzbtfl') LIMIT 21
2. SELECT "customuser_customuser"."password", "customuser_customuser"."last_login", "customuser_customuser"."is_superuser", "customuser_customuser"."identifier", "customuser_customuser"."username", "customuser_customuser"."email", "customuser_customuser"."is_staff", "customuser_customuser"."is_active", "customuser_customuser"."first_name", "customuser_customuser"."last_name", "customuser_customuser"."country", "customuser_customuser"."attachment" FROM "customuser_customuser" WHERE "customuser_customuser"."identifier" = 2026552186 LIMIT 21
3. SELECT "django_content_type"."app_label", "auth_permission"."codename" FROM "auth_permission" INNER JOIN "customuser_customuser_user_permissions" ON ("auth_permission"."id" = "customuser_customuser_user_permissions"."permission_id") INNER JOIN "django_content_type" ON ("auth_permission"."content_type_id" = "django_content_type"."id") WHERE "customuser_customuser_user_permissions"."customuser_id" = 2026552186
4. SELECT "django_content_type"."app_label", "auth_permission"."codename" FROM "auth_permission" INNER JOIN "auth_group_permissions" ON ("auth_permission"."id" = "auth_group_permissions"."permission_id") INNER JOIN "auth_group" ON ("auth_group_permissions"."group_id" = "auth_group"."id") INNER JOIN "customuser_customuser_groups" ON ("auth_group"."id" = "customuser_customuser_groups"."group_id") INNER JOIN "django_content_type" ON ("auth_permission"."content_type_id" = "django_content_type"."id") WHERE "customuser_customuser_groups"."customuser_id" = 2026552186
5. SELECT "wagtailusers_userprofile"."id", "wagtailusers_userprofile"."user_id", "wagtailusers_userprofile"."submitted_notifications", "wagtailusers_userprofile"."approved_notifications", "wagtailusers_userprofile"."rejected_notifications", "wagtailusers_userprofile"."updated_comments_notifications", "wagtailusers_userprofile"."preferred_language", "wagtailusers_userprofile"."current_time_zone", "wagtailusers_userprofile"."avatar", "wagtailusers_userprofile"."dismissibles", "wagtailusers_userprofile"."theme" FROM "wagtailusers_userprofile" WHERE "wagtailusers_userprofile"."user_id" = 2026552186 LIMIT 21
6. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."id" = 5 LIMIT 21
7. SELECT "wagtailcore_workflowstate"."id", "wagtailcore_workflowstate"."content_type_id", "wagtailcore_workflowstate"."base_content_type_id", "wagtailcore_workflowstate"."object_id", "wagtailcore_workflowstate"."workflow_id", "wagtailcore_workflowstate"."status", "wagtailcore_workflowstate"."created_at", "wagtailcore_workflowstate"."requested_by_id", "wagtailcore_workflowstate"."current_task_state_id", "wagtailcore_taskstate"."id", "wagtailcore_taskstate"."workflow_state_id", "wagtailcore_taskstate"."revision_id", "wagtailcore_taskstate"."task_id", "wagtailcore_taskstate"."status", "wagtailcore_taskstate"."started_at", "wagtailcore_taskstate"."finished_at", "wagtailcore_taskstate"."finished_by_id", "wagtailcore_taskstate"."comment", "wagtailcore_taskstate"."content_type_id", "wagtailcore_task"."id", "wagtailcore_task"."name", "wagtailcore_task"."content_type_id", "wagtailcore_task"."active" FROM "wagtailcore_workflowstate" LEFT OUTER JOIN "wagtailcore_taskstate" ON ("wagtailcore_workflowstate"."current_task_state_id" = "wagtailcore_taskstate"."id") LEFT OUTER JOIN "wagtailcore_task" ON ("wagtailcore_taskstate"."task_id" = "wagtailcore_task"."id") WHERE (("wagtailcore_workflowstate"."status" = 'in_progress' OR "wagtailcore_workflowstate"."status" = 'needs_changes') AND ("wagtailcore_workflowstate"."base_content_type_id" = 1 AND "wagtailcore_workflowstate"."object_id" IN ('5')))
8. SELECT "wagtailcore_revision"."id", "wagtailcore_revision"."content_type_id", "wagtailcore_revision"."base_content_type_id", "wagtailcore_revision"."object_id", "wagtailcore_revision"."submitted_for_moderation", "wagtailcore_revision"."created_at", "wagtailcore_revision"."user_id", "wagtailcore_revision"."object_str", "wagtailcore_revision"."content", "wagtailcore_revision"."approved_go_live_at" FROM "wagtailcore_revision" WHERE ("wagtailcore_revision"."content_type_id" = 38 AND "wagtailcore_revision"."object_id" = '5' AND "wagtailcore_revision"."approved_go_live_at" IS NOT NULL) ORDER BY "wagtailcore_revision"."id" ASC LIMIT 1
9. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id", "tests_eventpage"."page_ptr_id", "tests_eventpage"."date_from", "tests_eventpage"."date_to", "tests_eventpage"."time_from", "tests_eventpage"."time_to", "tests_eventpage"."audience", "tests_eventpage"."location", "tests_eventpage"."body", "tests_eventpage"."cost", "tests_eventpage"."signup_link", "tests_eventpage"."feed_image_id" FROM "tests_eventpage" INNER JOIN "wagtailcore_page" ON ("tests_eventpage"."page_ptr_id" = "wagtailcore_page"."id") WHERE "tests_eventpage"."page_ptr_id" = 5 LIMIT 21
10. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."path" = '00010001' LIMIT 21
11. SELECT "wagtailcore_grouppagepermission"."id", "wagtailcore_grouppagepermission"."group_id", "wagtailcore_grouppagepermission"."page_id", "wagtailcore_grouppagepermission"."permission_id", "wagtailcore_grouppagepermission"."permission_type", "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id", "auth_permission"."id", "auth_permission"."name", "auth_permission"."content_type_id", "auth_permission"."codename" FROM "wagtailcore_grouppagepermission" INNER JOIN "auth_group" ON ("wagtailcore_grouppagepermission"."group_id" = "auth_group"."id") INNER JOIN "customuser_customuser_groups" ON ("auth_group"."id" = "customuser_customuser_groups"."group_id") INNER JOIN "wagtailcore_page" ON ("wagtailcore_grouppagepermission"."page_id" = "wagtailcore_page"."id") LEFT OUTER JOIN "auth_permission" ON ("wagtailcore_grouppagepermission"."permission_id" = "auth_permission"."id") WHERE "customuser_customuser_groups"."customuser_id" = 2026552186
12. SELECT "wagtailcore_pagesubscription"."id", "wagtailcore_pagesubscription"."user_id", "wagtailcore_pagesubscription"."page_id", "wagtailcore_pagesubscription"."comment_notifications" FROM "wagtailcore_pagesubscription" WHERE ("wagtailcore_pagesubscription"."page_id" = 5 AND "wagtailcore_pagesubscription"."user_id" = 2026552186) LIMIT 21
13. SELECT "wagtailcore_workflowstate"."id", "wagtailcore_workflowstate"."content_type_id", "wagtailcore_workflowstate"."base_content_type_id", "wagtailcore_workflowstate"."object_id", "wagtailcore_workflowstate"."workflow_id", "wagtailcore_workflowstate"."status", "wagtailcore_workflowstate"."created_at", "wagtailcore_workflowstate"."requested_by_id", "wagtailcore_workflowstate"."current_task_state_id" FROM "wagtailcore_workflowstate" WHERE ("wagtailcore_workflowstate"."base_content_type_id" = 1 AND "wagtailcore_workflowstate"."object_id" = '5') ORDER BY "wagtailcore_workflowstate"."created_at" DESC LIMIT 1
14. SELECT "wagtailcore_comment"."id", "wagtailcore_comment"."page_id", "wagtailcore_comment"."user_id", "wagtailcore_comment"."text", "wagtailcore_comment"."contentpath", "wagtailcore_comment"."position", "wagtailcore_comment"."created_at", "wagtailcore_comment"."updated_at", "wagtailcore_comment"."revision_created_id", "wagtailcore_comment"."resolved_at", "wagtailcore_comment"."resolved_by_id" FROM "wagtailcore_comment" WHERE "wagtailcore_comment"."page_id" = 5
15. SELECT (1) AS "a" FROM "wagtailcore_page" INNER JOIN "wagtailcore_workflowpage" ON ("wagtailcore_page"."id" = "wagtailcore_workflowpage"."page_id") INNER JOIN "wagtailcore_workflow" ON ("wagtailcore_workflowpage"."workflow_id" = "wagtailcore_workflow"."id") WHERE ("wagtailcore_page"."path" IN ('0001', '00010001', '000100010003') AND "wagtailcore_workflowpage"."page_id" IS NOT NULL AND "wagtailcore_workflow"."active") LIMIT 1
16. SELECT (1) AS "a" FROM "wagtailcore_page" INNER JOIN "wagtailcore_workflowpage" ON ("wagtailcore_page"."id" = "wagtailcore_workflowpage"."page_id") INNER JOIN "wagtailcore_workflow" ON ("wagtailcore_workflowpage"."workflow_id" = "wagtailcore_workflow"."id") WHERE ("wagtailcore_page"."path" IN ('0001', '00010001', '000100010003') AND "wagtailcore_workflowpage"."page_id" IS NOT NULL AND "wagtailcore_workflow"."active") LIMIT 1
17. SELECT "tests_eventpagecarouselitem"."id", "tests_eventpagecarouselitem"."translation_key", "tests_eventpagecarouselitem"."locale_id", "tests_eventpagecarouselitem"."sort_order", "tests_eventpagecarouselitem"."link_external", "tests_eventpagecarouselitem"."link_page_id", "tests_eventpagecarouselitem"."link_document_id", "tests_eventpagecarouselitem"."image_id", "tests_eventpagecarouselitem"."embed_url", "tests_eventpagecarouselitem"."caption", "tests_eventpagecarouselitem"."page_id" FROM "tests_eventpagecarouselitem" WHERE "tests_eventpagecarouselitem"."page_id" = 5 ORDER BY "tests_eventpagecarouselitem"."sort_order" ASC
18. SELECT "tests_eventpagespeaker"."id", "tests_eventpagespeaker"."translation_key", "tests_eventpagespeaker"."locale_id", "tests_eventpagespeaker"."sort_order", "tests_eventpagespeaker"."link_external", "tests_eventpagespeaker"."link_page_id", "tests_eventpagespeaker"."link_document_id", "tests_eventpagespeaker"."page_id", "tests_eventpagespeaker"."first_name", "tests_eventpagespeaker"."last_name", "tests_eventpagespeaker"."image_id" FROM "tests_eventpagespeaker" WHERE "tests_eventpagespeaker"."page_id" = 5 ORDER BY "tests_eventpagespeaker"."sort_order" ASC
19. SELECT "tests_eventpagerelatedlink"."id", "tests_eventpagerelatedlink"."translation_key", "tests_eventpagerelatedlink"."locale_id", "tests_eventpagerelatedlink"."sort_order", "tests_eventpagerelatedlink"."link_external", "tests_eventpagerelatedlink"."link_page_id", "tests_eventpagerelatedlink"."link_document_id", "tests_eventpagerelatedlink"."title", "tests_eventpagerelatedlink"."page_id" FROM "tests_eventpagerelatedlink" WHERE "tests_eventpagerelatedlink"."page_id" = 5 ORDER BY "tests_eventpagerelatedlink"."sort_order" ASC
20. SELECT "tests_headcountrelatedmodelusingpk"."custom_id", "tests_headcountrelatedmodelusingpk"."event_page_id", "tests_headcountrelatedmodelusingpk"."head_count" FROM "tests_headcountrelatedmodelusingpk" WHERE "tests_headcountrelatedmodelusingpk"."event_page_id" = 5 ORDER BY "tests_headcountrelatedmodelusingpk"."custom_id" ASC
21. SELECT "wagtailcore_locale"."id", "wagtailcore_locale"."language_code" FROM "wagtailcore_locale" ORDER BY "wagtailcore_locale"."language_code" ASC
22. SELECT "wagtailcore_groupcollectionpermission"."id", "wagtailcore_groupcollectionpermission"."group_id", "wagtailcore_groupcollectionpermission"."collection_id", "wagtailcore_groupcollectionpermission"."permission_id", "wagtailcore_collection"."id", "wagtailcore_collection"."path", "wagtailcore_collection"."depth", "wagtailcore_collection"."numchild", "wagtailcore_collection"."name", "auth_permission"."id", "auth_permission"."name", "auth_permission"."content_type_id", "auth_permission"."codename" FROM "wagtailcore_groupcollectionpermission" INNER JOIN "auth_group" ON ("wagtailcore_groupcollectionpermission"."group_id" = "auth_group"."id") INNER JOIN "customuser_customuser_groups" ON ("auth_group"."id" = "customuser_customuser_groups"."group_id") INNER JOIN "wagtailcore_collection" ON ("wagtailcore_groupcollectionpermission"."collection_id" = "wagtailcore_collection"."id") INNER JOIN "auth_permission" ON ("wagtailcore_groupcollectionpermission"."permission_id" = "auth_permission"."id") WHERE "customuser_customuser_groups"."customuser_id" = 2026552186
23. SELECT (1) AS "a" FROM "wagtailcore_page" WHERE ((("wagtailcore_page"."path" LIKE '0001%' ESCAPE '\' AND "wagtailcore_page"."depth" >= 1 AND "wagtailcore_page"."owner_id" = 2026552186) OR ("wagtailcore_page"."path" LIKE '0001%' ESCAPE '\' AND "wagtailcore_page"."depth" >= 1)) AND "wagtailcore_page"."content_type_id" IN (189, 40, 79, 67, 78, 75, 65)) LIMIT 1
24. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."path" = '0001' LIMIT 21
25. SELECT "wagtailcore_page"."id", "wagtailcore_page"."content_type_id" FROM "wagtailcore_page" WHERE ("wagtailcore_page"."path" IN ('0001', '00010001', '000100010003') AND "wagtailcore_page"."path" LIKE '0001%' ESCAPE '\' AND "wagtailcore_page"."depth" >= 1) ORDER BY "wagtailcore_page"."path" ASC
26. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."id" IN (1, 2) ORDER BY "wagtailcore_page"."path" ASC
27. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id", "tests_eventpage"."page_ptr_id", "tests_eventpage"."date_from", "tests_eventpage"."date_to", "tests_eventpage"."time_from", "tests_eventpage"."time_to", "tests_eventpage"."audience", "tests_eventpage"."location", "tests_eventpage"."body", "tests_eventpage"."cost", "tests_eventpage"."signup_link", "tests_eventpage"."feed_image_id" FROM "tests_eventpage" INNER JOIN "wagtailcore_page" ON ("tests_eventpage"."page_ptr_id" = "wagtailcore_page"."id") WHERE "tests_eventpage"."page_ptr_id" IN (5) ORDER BY "wagtailcore_page"."path" ASC
28. SELECT (1) AS "a" FROM "wagtailcore_page" WHERE ("wagtailcore_page"."path" LIKE '000100010003%' ESCAPE '\' AND "wagtailcore_page"."depth" >= 3 AND "wagtailcore_page"."live") LIMIT 1
29. SELECT (1) AS "a" FROM "wagtailcore_page" WHERE ("wagtailcore_page"."path" LIKE '000100010003%' ESCAPE '\' AND "wagtailcore_page"."depth" >= 3 AND "wagtailcore_page"."live") LIMIT 1
30. SELECT "cache_key", "value", "expires" FROM "cache" WHERE "cache_key" IN (':2:wagtail_site_root_paths')
31. SELECT "wagtailcore_page"."path" FROM "wagtailcore_pageviewrestriction" INNER JOIN "wagtailcore_page" ON ("wagtailcore_pageviewrestriction"."page_id" = "wagtailcore_page"."id")
32. SELECT COUNT(*) FROM (SELECT DISTINCT "wagtailcore_referenceindex"."base_content_type_id" AS "col1", "wagtailcore_referenceindex"."object_id" AS "col2" FROM "wagtailcore_referenceindex" WHERE ("wagtailcore_referenceindex"."to_content_type_id" = 1 AND "wagtailcore_referenceindex"."to_object_id" = '5')) subquery
33. SELECT "tests_eventcategory"."id", "tests_eventcategory"."translation_key", "tests_eventcategory"."locale_id", "tests_eventcategory"."name" FROM "tests_eventcategory" INNER JOIN "tests_eventpage_categories" ON ("tests_eventcategory"."id" = "tests_eventpage_categories"."eventcategory_id") WHERE "tests_eventpage_categories"."eventpage_id" = 5
34. SELECT "tests_eventcategory"."id", "tests_eventcategory"."translation_key", "tests_eventcategory"."locale_id", "tests_eventcategory"."name" FROM "tests_eventcategory"
35. SELECT "customuser_customuser"."password", "customuser_customuser"."last_login", "customuser_customuser"."is_superuser", "customuser_customuser"."identifier", "customuser_customuser"."username", "customuser_customuser"."email", "customuser_customuser"."is_staff", "customuser_customuser"."is_active", "customuser_customuser"."first_name", "customuser_customuser"."last_name", "customuser_customuser"."country", "customuser_customuser"."attachment", "wagtailusers_userprofile"."id", "wagtailusers_userprofile"."user_id", "wagtailusers_userprofile"."submitted_notifications", "wagtailusers_userprofile"."approved_notifications", "wagtailusers_userprofile"."rejected_notifications", "wagtailusers_userprofile"."updated_comments_notifications", "wagtailusers_userprofile"."preferred_language", "wagtailusers_userprofile"."current_time_zone", "wagtailusers_userprofile"."avatar", "wagtailusers_userprofile"."dismissibles", "wagtailusers_userprofile"."theme" FROM "customuser_customuser" LEFT OUTER JOIN "wagtailusers_userprofile" ON ("customuser_customuser"."identifier" = "wagtailusers_userprofile"."user_id") WHERE "customuser_customuser"."identifier" IN (2026552186)
36. SELECT "wagtailcore_workflowpage"."page_id", "wagtailcore_workflowpage"."workflow_id" FROM "wagtailcore_workflowpage" WHERE "wagtailcore_workflowpage"."page_id" = 5 LIMIT 21
38. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" INNER JOIN "wagtailcore_workflowpage" ON ("wagtailcore_page"."id" = "wagtailcore_workflowpage"."page_id") INNER JOIN "wagtailcore_workflow" ON ("wagtailcore_workflowpage"."workflow_id" = "wagtailcore_workflow"."id") WHERE ("wagtailcore_page"."path" IN ('0001', '00010001', '000100010003') AND NOT ("wagtailcore_page"."id" = 5) AND "wagtailcore_workflowpage"."page_id" IS NOT NULL AND "wagtailcore_workflow"."active") ORDER BY "wagtailcore_page"."depth" DESC LIMIT 1
39. SELECT "wagtailcore_workflowpage"."page_id", "wagtailcore_workflowpage"."workflow_id" FROM "wagtailcore_workflowpage" WHERE "wagtailcore_workflowpage"."page_id" = 1 LIMIT 21
40. SELECT "wagtailcore_workflow"."id", "wagtailcore_workflow"."name", "wagtailcore_workflow"."active" FROM "wagtailcore_workflow" WHERE "wagtailcore_workflow"."id" = 1 LIMIT 21


======================================================================
test_page_edit_num_queries_as_superuser (wagtail.admin.tests.pages.test_edit_page.TestPageEdit.test_page_edit_num_queries_as_superuser)
----------------------------------------------------------------------

Captured queries were:
1. SELECT "django_session"."session_key", "django_session"."session_data", "django_session"."expire_date" FROM "django_session" WHERE ("django_session"."expire_date" > '2023-11-01 09:23:50.825925' AND "django_session"."session_key" = 'egp0fd1ooy15bvi7my5p3y6x4xeovqsj') LIMIT 21
2. SELECT "customuser_customuser"."password", "customuser_customuser"."last_login", "customuser_customuser"."is_superuser", "customuser_customuser"."identifier", "customuser_customuser"."username", "customuser_customuser"."email", "customuser_customuser"."is_staff", "customuser_customuser"."is_active", "customuser_customuser"."first_name", "customuser_customuser"."last_name", "customuser_customuser"."country", "customuser_customuser"."attachment" FROM "customuser_customuser" WHERE "customuser_customuser"."identifier" = -69225353 LIMIT 21
3. SELECT "wagtailusers_userprofile"."id", "wagtailusers_userprofile"."user_id", "wagtailusers_userprofile"."submitted_notifications", "wagtailusers_userprofile"."approved_notifications", "wagtailusers_userprofile"."rejected_notifications", "wagtailusers_userprofile"."updated_comments_notifications", "wagtailusers_userprofile"."preferred_language", "wagtailusers_userprofile"."current_time_zone", "wagtailusers_userprofile"."avatar", "wagtailusers_userprofile"."dismissibles", "wagtailusers_userprofile"."theme" FROM "wagtailusers_userprofile" WHERE "wagtailusers_userprofile"."user_id" = -69225353 LIMIT 21
4. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."id" = 5 LIMIT 21
5. SELECT "wagtailcore_workflowstate"."id", "wagtailcore_workflowstate"."content_type_id", "wagtailcore_workflowstate"."base_content_type_id", "wagtailcore_workflowstate"."object_id", "wagtailcore_workflowstate"."workflow_id", "wagtailcore_workflowstate"."status", "wagtailcore_workflowstate"."created_at", "wagtailcore_workflowstate"."requested_by_id", "wagtailcore_workflowstate"."current_task_state_id", "wagtailcore_taskstate"."id", "wagtailcore_taskstate"."workflow_state_id", "wagtailcore_taskstate"."revision_id", "wagtailcore_taskstate"."task_id", "wagtailcore_taskstate"."status", "wagtailcore_taskstate"."started_at", "wagtailcore_taskstate"."finished_at", "wagtailcore_taskstate"."finished_by_id", "wagtailcore_taskstate"."comment", "wagtailcore_taskstate"."content_type_id", "wagtailcore_task"."id", "wagtailcore_task"."name", "wagtailcore_task"."content_type_id", "wagtailcore_task"."active" FROM "wagtailcore_workflowstate" LEFT OUTER JOIN "wagtailcore_taskstate" ON ("wagtailcore_workflowstate"."current_task_state_id" = "wagtailcore_taskstate"."id") LEFT OUTER JOIN "wagtailcore_task" ON ("wagtailcore_taskstate"."task_id" = "wagtailcore_task"."id") WHERE (("wagtailcore_workflowstate"."status" = 'in_progress' OR "wagtailcore_workflowstate"."status" = 'needs_changes') AND ("wagtailcore_workflowstate"."base_content_type_id" = 1 AND "wagtailcore_workflowstate"."object_id" IN ('5')))
6. SELECT "wagtailcore_revision"."id", "wagtailcore_revision"."content_type_id", "wagtailcore_revision"."base_content_type_id", "wagtailcore_revision"."object_id", "wagtailcore_revision"."submitted_for_moderation", "wagtailcore_revision"."created_at", "wagtailcore_revision"."user_id", "wagtailcore_revision"."object_str", "wagtailcore_revision"."content", "wagtailcore_revision"."approved_go_live_at" FROM "wagtailcore_revision" WHERE ("wagtailcore_revision"."content_type_id" = 38 AND "wagtailcore_revision"."object_id" = '5' AND "wagtailcore_revision"."approved_go_live_at" IS NOT NULL) ORDER BY "wagtailcore_revision"."id" ASC LIMIT 1
7. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id", "tests_eventpage"."page_ptr_id", "tests_eventpage"."date_from", "tests_eventpage"."date_to", "tests_eventpage"."time_from", "tests_eventpage"."time_to", "tests_eventpage"."audience", "tests_eventpage"."location", "tests_eventpage"."body", "tests_eventpage"."cost", "tests_eventpage"."signup_link", "tests_eventpage"."feed_image_id" FROM "tests_eventpage" INNER JOIN "wagtailcore_page" ON ("tests_eventpage"."page_ptr_id" = "wagtailcore_page"."id") WHERE "tests_eventpage"."page_ptr_id" = 5 LIMIT 21
8. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."path" = '00010001' LIMIT 21
9. SELECT "wagtailcore_pagesubscription"."id", "wagtailcore_pagesubscription"."user_id", "wagtailcore_pagesubscription"."page_id", "wagtailcore_pagesubscription"."comment_notifications" FROM "wagtailcore_pagesubscription" WHERE ("wagtailcore_pagesubscription"."page_id" = 5 AND "wagtailcore_pagesubscription"."user_id" = -69225353) LIMIT 21
10. SELECT "wagtailcore_workflowstate"."id", "wagtailcore_workflowstate"."content_type_id", "wagtailcore_workflowstate"."base_content_type_id", "wagtailcore_workflowstate"."object_id", "wagtailcore_workflowstate"."workflow_id", "wagtailcore_workflowstate"."status", "wagtailcore_workflowstate"."created_at", "wagtailcore_workflowstate"."requested_by_id", "wagtailcore_workflowstate"."current_task_state_id" FROM "wagtailcore_workflowstate" WHERE ("wagtailcore_workflowstate"."base_content_type_id" = 1 AND "wagtailcore_workflowstate"."object_id" = '5') ORDER BY "wagtailcore_workflowstate"."created_at" DESC LIMIT 1
11. SELECT "wagtailcore_comment"."id", "wagtailcore_comment"."page_id", "wagtailcore_comment"."user_id", "wagtailcore_comment"."text", "wagtailcore_comment"."contentpath", "wagtailcore_comment"."position", "wagtailcore_comment"."created_at", "wagtailcore_comment"."updated_at", "wagtailcore_comment"."revision_created_id", "wagtailcore_comment"."resolved_at", "wagtailcore_comment"."resolved_by_id" FROM "wagtailcore_comment" WHERE "wagtailcore_comment"."page_id" = 5
12. SELECT (1) AS "a" FROM "wagtailcore_page" INNER JOIN "wagtailcore_workflowpage" ON ("wagtailcore_page"."id" = "wagtailcore_workflowpage"."page_id") INNER JOIN "wagtailcore_workflow" ON ("wagtailcore_workflowpage"."workflow_id" = "wagtailcore_workflow"."id") WHERE ("wagtailcore_page"."path" IN ('0001', '00010001', '000100010003') AND "wagtailcore_workflowpage"."page_id" IS NOT NULL AND "wagtailcore_workflow"."active") LIMIT 1
13. SELECT (1) AS "a" FROM "wagtailcore_page" INNER JOIN "wagtailcore_workflowpage" ON ("wagtailcore_page"."id" = "wagtailcore_workflowpage"."page_id") INNER JOIN "wagtailcore_workflow" ON ("wagtailcore_workflowpage"."workflow_id" = "wagtailcore_workflow"."id") WHERE ("wagtailcore_page"."path" IN ('0001', '00010001', '000100010003') AND "wagtailcore_workflowpage"."page_id" IS NOT NULL AND "wagtailcore_workflow"."active") LIMIT 1
14. SELECT "tests_eventpagecarouselitem"."id", "tests_eventpagecarouselitem"."translation_key", "tests_eventpagecarouselitem"."locale_id", "tests_eventpagecarouselitem"."sort_order", "tests_eventpagecarouselitem"."link_external", "tests_eventpagecarouselitem"."link_page_id", "tests_eventpagecarouselitem"."link_document_id", "tests_eventpagecarouselitem"."image_id", "tests_eventpagecarouselitem"."embed_url", "tests_eventpagecarouselitem"."caption", "tests_eventpagecarouselitem"."page_id" FROM "tests_eventpagecarouselitem" WHERE "tests_eventpagecarouselitem"."page_id" = 5 ORDER BY "tests_eventpagecarouselitem"."sort_order" ASC
15. SELECT "tests_eventpagespeaker"."id", "tests_eventpagespeaker"."translation_key", "tests_eventpagespeaker"."locale_id", "tests_eventpagespeaker"."sort_order", "tests_eventpagespeaker"."link_external", "tests_eventpagespeaker"."link_page_id", "tests_eventpagespeaker"."link_document_id", "tests_eventpagespeaker"."page_id", "tests_eventpagespeaker"."first_name", "tests_eventpagespeaker"."last_name", "tests_eventpagespeaker"."image_id" FROM "tests_eventpagespeaker" WHERE "tests_eventpagespeaker"."page_id" = 5 ORDER BY "tests_eventpagespeaker"."sort_order" ASC
16. SELECT "tests_eventpagerelatedlink"."id", "tests_eventpagerelatedlink"."translation_key", "tests_eventpagerelatedlink"."locale_id", "tests_eventpagerelatedlink"."sort_order", "tests_eventpagerelatedlink"."link_external", "tests_eventpagerelatedlink"."link_page_id", "tests_eventpagerelatedlink"."link_document_id", "tests_eventpagerelatedlink"."title", "tests_eventpagerelatedlink"."page_id" FROM "tests_eventpagerelatedlink" WHERE "tests_eventpagerelatedlink"."page_id" = 5 ORDER BY "tests_eventpagerelatedlink"."sort_order" ASC
17. SELECT "tests_headcountrelatedmodelusingpk"."custom_id", "tests_headcountrelatedmodelusingpk"."event_page_id", "tests_headcountrelatedmodelusingpk"."head_count" FROM "tests_headcountrelatedmodelusingpk" WHERE "tests_headcountrelatedmodelusingpk"."event_page_id" = 5 ORDER BY "tests_headcountrelatedmodelusingpk"."custom_id" ASC
18. SELECT "wagtailcore_locale"."id", "wagtailcore_locale"."language_code" FROM "wagtailcore_locale" ORDER BY "wagtailcore_locale"."language_code" ASC
19. SELECT (1) AS "a" FROM "wagtailcore_page" WHERE "wagtailcore_page"."content_type_id" IN (189, 40, 79, 67, 78, 75, 65) LIMIT 1
20. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."depth" = 1 ORDER BY "wagtailcore_page"."path" ASC
21. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."path" = '0001' LIMIT 21
22. SELECT "wagtailcore_page"."id", "wagtailcore_page"."content_type_id" FROM "wagtailcore_page" WHERE ("wagtailcore_page"."path" IN ('0001', '00010001', '000100010003') AND "wagtailcore_page"."path" LIKE '0001%' ESCAPE '\' AND "wagtailcore_page"."depth" >= 1) ORDER BY "wagtailcore_page"."path" ASC
23. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" WHERE "wagtailcore_page"."id" IN (1, 2) ORDER BY "wagtailcore_page"."path" ASC
24. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id", "tests_eventpage"."page_ptr_id", "tests_eventpage"."date_from", "tests_eventpage"."date_to", "tests_eventpage"."time_from", "tests_eventpage"."time_to", "tests_eventpage"."audience", "tests_eventpage"."location", "tests_eventpage"."body", "tests_eventpage"."cost", "tests_eventpage"."signup_link", "tests_eventpage"."feed_image_id" FROM "tests_eventpage" INNER JOIN "wagtailcore_page" ON ("tests_eventpage"."page_ptr_id" = "wagtailcore_page"."id") WHERE "tests_eventpage"."page_ptr_id" IN (5) ORDER BY "wagtailcore_page"."path" ASC
25. SELECT (1) AS "a" FROM "wagtailcore_locale" WHERE NOT ("wagtailcore_locale"."id" IN (SELECT U1."locale_id" FROM "tests_eventpage" U0 INNER JOIN "wagtailcore_page" U1 ON (U0."page_ptr_id" = U1."id") WHERE U1."translation_key" = 'd987b6f5901e404aa8f1eba3676e391b')) LIMIT 1
26. SELECT "cache_key", "value", "expires" FROM "cache" WHERE "cache_key" IN (':2:wagtail_site_root_paths')
27. SELECT "wagtailcore_page"."path" FROM "wagtailcore_pageviewrestriction" INNER JOIN "wagtailcore_page" ON ("wagtailcore_pageviewrestriction"."page_id" = "wagtailcore_page"."id")
28. SELECT COUNT(*) FROM (SELECT DISTINCT "wagtailcore_referenceindex"."base_content_type_id" AS "col1", "wagtailcore_referenceindex"."object_id" AS "col2" FROM "wagtailcore_referenceindex" WHERE ("wagtailcore_referenceindex"."to_content_type_id" = 1 AND "wagtailcore_referenceindex"."to_object_id" = '5')) subquery
29. SELECT "tests_eventcategory"."id", "tests_eventcategory"."translation_key", "tests_eventcategory"."locale_id", "tests_eventcategory"."name" FROM "tests_eventcategory" INNER JOIN "tests_eventpage_categories" ON ("tests_eventcategory"."id" = "tests_eventpage_categories"."eventcategory_id") WHERE "tests_eventpage_categories"."eventpage_id" = 5
30. SELECT "tests_eventcategory"."id", "tests_eventcategory"."translation_key", "tests_eventcategory"."locale_id", "tests_eventcategory"."name" FROM "tests_eventcategory"
31. SELECT "customuser_customuser"."password", "customuser_customuser"."last_login", "customuser_customuser"."is_superuser", "customuser_customuser"."identifier", "customuser_customuser"."username", "customuser_customuser"."email", "customuser_customuser"."is_staff", "customuser_customuser"."is_active", "customuser_customuser"."first_name", "customuser_customuser"."last_name", "customuser_customuser"."country", "customuser_customuser"."attachment", "wagtailusers_userprofile"."id", "wagtailusers_userprofile"."user_id", "wagtailusers_userprofile"."submitted_notifications", "wagtailusers_userprofile"."approved_notifications", "wagtailusers_userprofile"."rejected_notifications", "wagtailusers_userprofile"."updated_comments_notifications", "wagtailusers_userprofile"."preferred_language", "wagtailusers_userprofile"."current_time_zone", "wagtailusers_userprofile"."avatar", "wagtailusers_userprofile"."dismissibles", "wagtailusers_userprofile"."theme" FROM "customuser_customuser" LEFT OUTER JOIN "wagtailusers_userprofile" ON ("customuser_customuser"."identifier" = "wagtailusers_userprofile"."user_id") WHERE "customuser_customuser"."identifier" IN (-69225353)
32. SELECT "wagtailcore_workflowpage"."page_id", "wagtailcore_workflowpage"."workflow_id" FROM "wagtailcore_workflowpage" WHERE "wagtailcore_workflowpage"."page_id" = 5 LIMIT 21
33. SELECT "wagtailcore_page"."id", "wagtailcore_page"."path", "wagtailcore_page"."depth", "wagtailcore_page"."numchild", "wagtailcore_page"."translation_key", "wagtailcore_page"."locale_id", "wagtailcore_page"."latest_revision_id", "wagtailcore_page"."live", "wagtailcore_page"."has_unpublished_changes", "wagtailcore_page"."first_published_at", "wagtailcore_page"."last_published_at", "wagtailcore_page"."live_revision_id", "wagtailcore_page"."go_live_at", "wagtailcore_page"."expire_at", "wagtailcore_page"."expired", "wagtailcore_page"."locked", "wagtailcore_page"."locked_at", "wagtailcore_page"."locked_by_id", "wagtailcore_page"."title", "wagtailcore_page"."draft_title", "wagtailcore_page"."slug", "wagtailcore_page"."content_type_id", "wagtailcore_page"."url_path", "wagtailcore_page"."owner_id", "wagtailcore_page"."seo_title", "wagtailcore_page"."show_in_menus", "wagtailcore_page"."search_description", "wagtailcore_page"."latest_revision_created_at", "wagtailcore_page"."alias_of_id" FROM "wagtailcore_page" INNER JOIN "wagtailcore_workflowpage" ON ("wagtailcore_page"."id" = "wagtailcore_workflowpage"."page_id") INNER JOIN "wagtailcore_workflow" ON ("wagtailcore_workflowpage"."workflow_id" = "wagtailcore_workflow"."id") WHERE ("wagtailcore_page"."path" IN ('0001', '00010001', '000100010003') AND NOT ("wagtailcore_page"."id" = 5) AND "wagtailcore_workflowpage"."page_id" IS NOT NULL AND "wagtailcore_workflow"."active") ORDER BY "wagtailcore_page"."depth" DESC LIMIT 1
34. SELECT "wagtailcore_workflowpage"."page_id", "wagtailcore_workflowpage"."workflow_id" FROM "wagtailcore_workflowpage" WHERE "wagtailcore_workflowpage"."page_id" = 1 LIMIT 21
35. SELECT "wagtailcore_workflow"."id", "wagtailcore_workflow"."name", "wagtailcore_workflow"."active" FROM "wagtailcore_workflow" WHERE "wagtailcore_workflow"."id" = 1 LIMIT 21
```
</details>

Reply:
Interesting, that means there is a duplicated query that we might be able to remove.

Reply:
I am not sure where to start with finding this, any tips?

I can see we are calling it before the `wagtailcore_groupcollectionpermission` the first time.

As for this new addition, is it possible there is a difference in when we call the template tag from within the template? Is this cached differently?

Is it possible that the original template one was never tracked in this number?

Reply:
Using Django debug toolbar and looking for the same query reveals that this template tag is used in the preview panel's template: https://github.com/wagtail/wagtail/blob/cdac7dc7f5defb194ab8ec7a0cc76a169ceff8de/wagtail/admin/templates/wagtailadmin/shared/side_panels/preview.html#L3

Previously the tag doesn't do any queries. Now it does, and we call this twice: once in this template and once in the `admin_base.html` template. The previous PR was made before the permissions rework that added these tests, so we didn't catch this.

I think we can either:

- Cache the result of the tag on the request object (or in the context, but this makes the cache unusable in template components which have their own context), and reuse it if the cache exists.
- Refactor `preview.html` so that it doesn't use the template tag, by either
  - Accessing the settings value directly in [PreviewSidePanel.get_context_data()](https://github.com/wagtail/wagtail/blob/cdac7dc7f5defb194ab8ec7a0cc76a169ceff8de/wagtail/admin/ui/side_panels.py#L319) and put it in the context.
  - Always rendering the refresh button, initially hidden but show it on JS init if auto update is disabled.

Reply:
Ahh brilliant. Thank you. I'll take another look.

Reply:
OK I think I have solved this, hopefully with still a small surface area of changes.

I opted for pulling out the auto preview settings from the global config and just preparing this in the side preview panel only. I do not think we really need these settings globally as already called out on https://github.com/wagtail/wagtail/issues/9771

This part has been rebased/prepared as the first commit on this PR with the rest of the config change as the second commit.

I have not cached the `wagtail_config` template tag, I was wondering if there is a way to somehow make this not a template tag though, it's really only suitable to be used once in the code. I thought I am better to leave this kind of refactoring to a later PR though.

Assuming the CI runs, this should hopefully not change any of the query counts anymore and leaves us in a position that's closer to where we probably want to go with making preview panel a Stimulus component.

Thanks for your help here Sage. It's been good to do a bit of digging into how to check for query counts and understand that better.